From e48cee6cca742a5bdb3daf6fa0b7c76073217faa Mon Sep 17 00:00:00 2001
From: Jekyll Wu <adaptee@gmail.com>
Date: Thu, 1 Nov 2012 11:57:42 +0800
Subject: [PATCH] No need to lookup executable PATH when absolute path is
 given (cherry picked from commit
 d12aead588ec09359633eb9de5f9ac2b2a7e1e06)

---
 src/Session.cpp |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Session.cpp b/src/Session.cpp
index 732858d..cf325b0 100644
--- a/src/Session.cpp
+++ b/src/Session.cpp
@@ -369,6 +369,11 @@ QString Session::checkProgram(const QString& program)
     if (exec.isEmpty())
         return QString();
 
+    QFileInfo info(exec);
+    if (info.isAbsolute() && info.exists() && info.isExecutable()) {
+        return exec;
+    }
+
     exec = KRun::binaryName(exec, false);
     exec = KShell::tildeExpand(exec);
     QString pexec = KStandardDirs::findExe(exec);
-- 
1.7.10.4

